lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -1722,21 +1722,28 @@ bool PullbackCloner::Implementation::run() {
   }
 
   // Create pullback blocks and arguments, visiting original blocks in
-  // post-order post-dominance order.
-  SmallVector<SILBasicBlock *, 8> postOrderPostDomOrder;
-  // Start from the root node, which may have a marker `nullptr` block if
-  // there are multiple roots.
-  PostOrderPostDominanceOrder postDomOrder(postDomInfo->getRootNode(),
-                                           postOrderInfo, original.size());
-  while (auto *origNode = postDomOrder.getNext()) {
-    auto *origBB = origNode->getBlock();
-    postDomOrder.pushChildren(origNode);
-    // If node is the `nullptr` marker basic block, do not push it.
-    if (!origBB)
-      continue;
-    postOrderPostDomOrder.push_back(origBB);
+  // dominance order.
+  SmallVector<SILBasicBlock *, 8> dominanceOrder = {};
+  std::deque<SILBasicBlock *> queue = {};
+  SmallDenseSet<SILBasicBlock *, 8> visitedBlocks = {};
```

Comment:
Minor naming and style suggestions:
```suggestion
  // The original blocks in traversal order for pullback generation.
  SmallVector<SILBasicBlock *, 8> originalBlocks;
  // The set of visited original blocks.
  SmallDenseSet<SILBasicBlock *, 8> visitedOriginalBlocks;
  
  // Perform BFS from the original exit block.
  {
    std::deque<SILBasicBlock *> worklist;
    ...
  }
```